### PR TITLE
Feature edit

### PR DESCRIPTION
**Pull request by jay-tyler**

### CODE REVIEW

**Comment by jay-tyler** [comment]
This closes #10

File: requirements.txt [hunk 1]
```diff
@@ -3,25 +3,47 @@ cffi==1.1.2
...
 ipaddress==1.0.7
+ipython==3.2.0
 Jinja2==2.7.3
+lettuce==0.2.20
+linecache2==1.0.0
+Mako==1.0.1
+Markdown==2.6.2
+markdown2==2.3.0
 MarkupSafe==0.23
+misaka==1.0.2
```
Comment (cewing) [inline requirements.txt:19]: I had to look this library up, and it appears to be a markdown rendering library that was last updated about three years ago.  I suspect that some of the other libraries I see in this list are also related and probably should not be used.  One of the tough aspects of using external libraries is the burden of tracking what you are using and what you are not using.  But you need to do this and then clean out packages that are not actually "required" from your `requirements.txt` file.

File: tests/test_journal.py [hunk 2]
```diff
@@ -184,13 +121,26 @@ def test_post_to_add_view_authorized(app, auth_req):
...
+    auth_req.params = {'username': 'admin', 'password': 'secret'}
+    redirect = app.post('/login', params=auth_req.params)
+    response = redirect.follow()
+    assert response.status_code == 200
+    entry_data = {
+        'body_text': 'This is a post',
+        'title': ''
+    }
+    response = app.post('/new', params=entry_data, status='2*')
+    assert 'Try Again: need both an entry and a title' in response.body
```
Comment (cewing) [inline tests/test_journal.py:134]: good test.

File: tests/test_journal_bdd.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+@then('The entry text should have markdown styling')
+def check_markdown(one_entry, app):
+    detail = app.get('/detail/{entry_id}'.format(entry_id=one_entry.id))
+    print detail.body
+    assert '<div class="markdown">' in detail.body
+    assert "<h1>Here is a large title, should be h1</h1>" in detail.body
+    assert "<h3>Here is a small title, should be h3</h3>" in detail.body
+    assert "<ul>" in detail.body
+    assert "<ol>" in detail.body
+    assert "<li>One</li>" in detail.body
```
Comment (cewing) [inline tests/test_journal_bdd.py:139]: remember that there are better ways to inspect the HTML that is generated by your app.  The response object returned by the `WebTest` `TestApp` object will have the `app.html` attribute that returns a `BeautifulSoup` object you can use to directly inspect the DOM without needing to do rough string comparisons.  This approach is a bit more robust to flaky layout changes.

File: tests/views.feature [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+Scenario: Checking Code Highlighting
+    Given I am an anonymous user
+    And There is an entry
+    Then The code text should have highlighting style
+
+
+Scenario: Check Markdown Styling
+    Given I am an anonymous user
+    And There is an entry
+    Then The entry text should have markdown styling
```
Comment (cewing) [inline tests/views.feature:29]: A good collection of features.

**Comment by cewing** [comment]
I don't appear to have any code here in this PR that covers the actual edit view(s), or the templates that you use on the front end.  Can you work on getting me a PR with that code so I can review it too?

**Comment by cewing** [comment]
Poop.  Wait a moment and cancel that last request.  I am now noticing that this is a resubmission directly addressing the issue of tests.  I apologize for the confusion.

**Comment by jay-tyler** [comment]
I've submitted this twice, one for tests, the other for the edit view. I'll
work on getting a PR that reflects the edit view wor

**Pull request merged**